### ISSUE TITLE
win_owner fails with recurse option specified

### ISSUE THREAD

**Issue by Vye** [ansible/ansible-modules-extras#2932]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

win_owner
##### ANSIBLE VERSION

```
ansible 2.1.1.0
```
##### CONFIGURATION

N/A
##### OS / ENVIRONMENT

Windows
##### SUMMARY

Ansible converts yes/true/no/false to booleans before passing them to win_owner but win_owner's `recurse` parameter seems is expecting yes/no string instead of a boolean.
##### STEPS TO REPRODUCE

```
- win_owner:
    path: "{{ app_dir }}"
    user: user@domain
    recurse: yes
```

##### EXPECTED RESULTS

Set ownership of specified directory recursively without errors.
##### ACTUAL RESULTS

```
fatal: [hostname]: FAILED! => {"changed": false, "failed": true, "msg": "Argument recurse needs to be one of no,yes but was True."}
```

**Comment by ansibot** [comment]
Maintainer unknown ping @gregdek @resmo @robynbergeron.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jhawkesworth** [comment]
I took a look at the code and this line is the culprit.
`
$recurse = Get-Attr $params "recurse" "no" -validateSet "no","yes" -resultobj $result`

I don't think we made it clear that True and False are interchangeable with Yes and No respectively in the rest of ansible in the module developers guide at the time this module was written.

I think a nice way to fix this would be to add a -validateBoolean option to Get-Attr, which would check if the parameter value was something that ansible considers to be a boolean (and return a boolean so module developers didn't have to do this sort of thing in the module code):

`$recurse = $recurse | ConvertTo-Bool`

At least there is an easy workaround to just modify playbook for now.

**Comment by Vye** [comment]
I was able to resolve it by patching win_owner.ps1 which is good enough for me until the next release. 

I don't know what you mean about modifying the playbook as a workaround though.

**Comment by jhawkesworth** [comment]
sorry I misunderstood.

You might be able to workaround by using key=value pairs for args in your playbook.

`- win_owner: "path: {{ app_dir }} user=user@domain recurse=yes"`

This is because using the key=value syntax doesn't preseve the yaml type information, so I think in this case the value of recurse would be passed to the module as a string.

Better to fix the module (and API) preferably, of course.

**Comment by Vye** [comment]
oh ok. I was patching Ansible to get around this but that's a better workaround until the bug is fixed. Thanks!

**Comment by ansibot** [comment]
@h0nIg, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by h0nIg** [comment]
this is an extra module, but it is not working on core modules too. therefore this is not intended and it seems that some yml feature is converting this value. 

https://github.com/ansible/ansible-modules-core/blob/devel/windows/win_lineinfile.ps1#L196
https://github.com/ansible/ansible-modules-core/blob/devel/windows/win_user.ps1#L77

currently i do not have much time to fix this, feel free to open a pull request or open a ticket on ansible/ ansible-modules-core to introduce the feature @jhawkesworth suggested

btw the following is working too:
`- win_owner: path="{{ app_dir }}" user=user recurse=yes`

**Comment by h0nIg** [comment]
wontfix "needs api extension"
